### PR TITLE
Switch to GHA CI workflow

### LINKED ISSUE #41: Switch to GHA CI workflow
Switch from Travis-CI to GHA CI workflow, as documented at https://gist.github.com/weierophinney/b003e50c3c2667d08076caf31ebd36a4

### PR DESCRIPTION
**Pull request by internalsystemerror**
Switch from Travis CI to use GitHub Actions according to: https://gist.github.com/weierophinney/b003e50c3c2667d08076caf31ebd36a4.

This closes #41

### CODE REVIEW

**Comment by internalsystemerror** [comment]
I'm getting different results locally compared to here ... ~~I actually pass CS checks locally since `phpcs.xml` is used whereas here it seems to be using `phpcs.xml.dist` instead (just a hypothesis)~~ (edit: fixed). And I'm also failing a lot more tests locally, feel free to take over this PR as needed otherwise I'll check back on this later.

File: test/MezzioInstallerTest/HomePageResponseTest.php [hunk 1]
```diff
@@ -218,7 +229,7 @@ public function installCasesProvider(): Generator
                     continue;
                 }
 
-                foreach ($this->intallTypes as $intallType) {
+                foreach ($this->installTypes as $intallType) {
```
Comment (brayniverse) [inline test/MezzioInstallerTest/HomePageResponseTest.php]: You could correct the `$intallType` typos here while you're fixing `$this->intallType`.

```suggestion
                foreach ($this->installTypes as $installType) {
```

File: test/MezzioInstallerTest/HomePageResponseTest.php [hunk 2]
```diff
@@ -307,7 +318,7 @@ public function rendererlessInstallCasesProvider(): Generator
                 $routerName   = $this->expectedRouterAttributes[$routerClass]['routerName'];
                 $routerDocs   = $this->expectedRouterAttributes[$routerClass]['routerDocs'];
 
-                foreach ($this->intallTypes as $intallType) {
+                foreach ($this->installTypes as $intallType) {
```
Comment (brayniverse) [inline test/MezzioInstallerTest/HomePageResponseTest.php]:
```suggestion
                foreach ($this->installTypes as $installType) {
```

The typo is here as well.

**Review by brayniverse — CHANGES_REQUESTED** [review]

**Comment by internalsystemerror** [comment]
I believe this is now ready for review

**Pull request merged**